benches/elgamal_benchmarks.rs

```diff
@@ -0,0 +1,131 @@
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+
+#[allow(dead_code)]
+mod elgamal_benches {
+    use super::*;
+    use curv::BigInt;
+    use elgamal::rfc7919_groups::SupportedGroups;
+    use elgamal::{ElGamal, ElGamalKeyPair, ElGamalPP};
+
+    static RFC_GROUPS: [SupportedGroups; 5] = [
+        SupportedGroups::FFDHE2048,
+        SupportedGroups::FFDHE3072,
+        SupportedGroups::FFDHE4096,
+        SupportedGroups::FFDHE6144,
+        SupportedGroups::FFDHE8192,
+    ];
+    static BIT_PARAMS: [usize; 2] = [1024, 2048];
+    static MSG_PARAMS: [u32; 2] = [9, 3];
+
+    fn keypair_from_rfc7919(group_id: SupportedGroups) -> ElGamalKeyPair {
+        let p_point = ElGamalPP::generate_from_rfc7919(group_id);
+        let keypair = ElGamalKeyPair::generate(&p_point);
+        keypair
+    }
+
+    fn keypair_from_bits(bit_size: usize) -> ElGamalKeyPair {
+        let p_point = ElGamalPP::generate(bit_size);
+        let keypair = ElGamalKeyPair::generate(&p_point);
+        keypair
+    }
+
+    fn elgamal_multiply(bit_size: usize, msg_params: &[u32; 2]) {
+        let pp = ElGamalPP::generate(bit_size);
+        let keypair = ElGamalKeyPair::generate(&pp);
+        let message1 = BigInt::from(msg_params[0]);
+        let c1 = ElGamal::encrypt(&message1, &keypair.pk).unwrap();
+        let message2 = BigInt::from(msg_params[1]);
+        let c2 = ElGamal::encrypt(&message2, &keypair.pk).unwrap();
+        let c = ElGamal::mul(&c1, &c2).unwrap();
+        let _message_tag = ElGamal::decrypt(&c, &keypair.sk).unwrap();
+    }
+
+    fn elgamal_power(group_id: SupportedGroups, msg_params: &[u32; 2]) {
```

Comment:
Same as before, it would be more informative if the bench will be limited only to the power.